### ISSUE TITLE
Bug in `[0] * sum([0])`

### ISSUE THREAD

**Issue by tamuhey** [microsoft/pyright#1715]
`list[int] * sum(list[int])` raises error:

```python
def f(x: list[int]) -> list[int]:
    return [0] * sum(
        x
    )  # error: Operator "*" not supported for types "list[int]" and "list[int] | int"


def g(x: list[int]) -> list[int]:
    y = sum(x)
    return [0] * y  # No error
```

pyright 1.1.127

**Comment by erictraut** [comment]
Thanks for the bug report. This was a regression that I introduced when adding this capability: https://github.com/microsoft/pyright/issues/1691.

This is why I'm always reluctant to add special cases like this because they don't generalize well. I've made the special case even more specific (limiting it to the case where the RHS is an integer literal only) to fix this problem.

**Comment by erictraut** [comment]
This fix is included in pyright 1.1.128, which I just published. It will also be included in the next release of pylance.
